sources/platform/actors/development/actor_definition/dataset_schema/validation.md

```diff
@@ -0,0 +1,312 @@
+            "type": "string"
+        }
+    },
+    "additionalProperties": true
+}
+```
+
+See [json schema reference](https://json-schema.org/understanding-json-schema/reference) for additional options.
+
+Example of schema generator [here](https://www.liquid-technologies.com/online-json-to-schema-converter).
+
+# Dataset field statistics
+
+When you have the dataset fields schema set up, we then use the schema to generate a list of fields and measure statistics for these fields.
+
+The measured statistics are following:
+
+- **Null count:** how many items in the dataset have the field set to null
+- **Empty count:** how many items in the dataset are `undefined` , meaning that for example empty string is not considered empty
+- **Minimum and maximum**
+  - For numbers, this is calculated directly
+  - For strings, this field tracks string length
+  - For arrays, this field tracks the number of items in the array
+  - For objects, this tracks the number of keys
+
+:::note
+
+Currently, you cannot view these statistics. We will add API endpoint soon. But you can already use them in monitoring.
+
+:::
+
+## Examples
+
+For this schema:
+
+```json
+{
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "type": "object",
+    "properties": {
+        "name": {
+            "type": "string"
+        },
+        "description": {
+            "type": "string"
+        },
+        "dimensions": {
+            "type": "object",
+            "nullable": true,
+            "properties": {
+                "width": {
+                    "type": "number"
+                },
+                "height": {
+                    "type": "number"
+                }
+            },
+            "required": ["width", "height"]
+        },
+        "price": {
+            "type": ["string", "number"]
+        }
+    },
+    "required": ["name", "price"]
+}
+```
+
+The stored statistics and fields in the database look like this:
+
+```json
+{
+    "_id" : "1lVGVBkWIhSYPY1dD",
+    "fields" : [
+        "name",
+        "description",
+        "dimensions",
+        "dimensions/width",
+        "dimensions/height",
+        "price"
+    ],
+    "stats": {
+        "description": {
+            "emptyCount": 105,
+            "max": 19,
+            "min": 19
+        },
+        "dimensions": {
+            "emptyCount": 144,
+            "max": 2,
+            "min": 2,
+            "nullCount": 86
+        },
+        "dimensions/height": {
+            "emptyCount": 230,
+            "max": 992,
+            "min": 18
+        },
+        "dimensions/width": {
+            "emptyCount": 230,
+            "max": 977,
+            "min": 4
+        },
+        "name": {
+            "max": 13,
+            "min": 11
+        },
+        "price": {
+            "max": 999,
+            "min": 1
+        }
+    }
+}
+```
+
+:::note
+
+If you want to see for yourself, check `datasetStatistics` collection. The ids correspond to the ids of datasets.
```

Comment:
This whole section should not be in the documentation. It should be added once we add the dataset statistics API endpoint with the link to the documentation of the endpoint.

Reply:
thanks for checking it, the example part is removed now